src/validation.cpp

```diff
@@ -2844,8 +2844,8 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
 
 bool IsWitnessEnabled(const CBlockIndex* pindexPrev, const Consensus::Params& params)
 {
-    LOCK(cs_main);
-    return (VersionBitsState(pindexPrev, params, Consensus::DEPLOYMENT_SEGWIT, versionbitscache) == THRESHOLD_ACTIVE);
+    int nHeight = pindexPrev == nullptr ? 0 : pindexPrev->nHeight + 1;
```

Comment:
nit: no hungarian notation.

You could just do away with the local variable entirely:

```
    return ((int)chainActive.Height() >= params.SEGWITHeight);
```

Reply:
Many different places use IsWitnessEnabled but not all look at the tip